### ISSUE TITLE
Smarty lexer issue

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#746]
(Original issue 1039 created by None on 2014-10-06T05:31:57.070566+00:00)

The Smarty template lexer doesn't recognize nested braces:

```
#!smarty

{templateFunction param={anotherFunction} param2=$something}
```
is valid syntax, but anything beyond the first (inner) closing brace will escape from Smarty. So the Smarty lexer should duplicate the code for entering Smarty state within this state.

**Comment by Anteru** [comment]
(Original issue 1039 last updated on 2014-10-07T03:13:58.083214+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-10-07T03:13:58.079677+00:00)

SmartyLexer: Support nested curly braces.

Fixes #1039

→ <<cset 0101e6d60d1b>>
